logistics/scripts/Invoke-SdkGen.ps1

```diff
@@ -74,29 +76,36 @@ function Invoke-SdkGen {
-                    $script:result += Invoke-Task "Invoke-SdkGenConsole" { Invoke-SdkGenConsole $apiMetadataUrl $buildConfiguration }
-                }
-                finally {
-                    $script:result += Invoke-Task -name "Stop-TestHarness" -task { Stop-TestHarness }
-                }
-
-                Write-Host "Updating package id and title to EdFi$suffix.OdsApi.TestSdk"
-                (Get-Content -path $nuspecFile -Raw) | ForEach-Object {
-                    $_.replace("<id>EdFi$suffix.OdsApi.Sdk</id>","<id>EdFi$suffix.OdsApi.TestSdk</id>").replace("<title>EdFi$suffix.OdsApi.Sdk</title>","<title>EdFi$suffix.OdsApi.TestSdk</title>")
-                } | Set-Content -Path $nuspecFile
-                
-                $script:result += Invoke-Task "Restore-ApiSdk-Packages" { Invoke-Restore-ApiSdk-Packages $sdkSolutionFile }
-                $script:result += Invoke-Task "Invoke-RebuildSolution" { Invoke-RebuildSolution $buildConfiguration "minimal" $sdkSolutionFile }
-                $script:result += Invoke-Task "Pack-ApiTestSdk" { Invoke-Pack-ApiSdk $buildConfiguration $teamCityParameters $version }
+                $script:result += Invoke-Task "Invoke-RebuildSolution" { Invoke-RebuildSolution $buildConfiguration "minimal"  $sdkGenSolution }
+                $script:result += Invoke-Task -name "Start-TestHarness" -task { Start-TestHarness $apiUrl $configurationFile $environmentFilePath $null "EdFiOdsApiSdk" }
+                $sdkCliVersion = Get-ValueOrDefault $teamCityParameters['SdkCliVersion'] '6.0.1'
+                $arguments = @("-v",$sdkCliVersion,"-p","-c","-i")
+                $script:result += Invoke-Task "Invoke-SdkGenConsole" { Invoke-SdkGenConsole $apiMetadataUrl $buildConfiguration $arguments }
             }
+            finally {
+                $script:result += Invoke-Task -name "Stop-TestHarness" -task { Stop-TestHarness }
+            }
+
+            $sdkSolutionFile = (Get-RepositoryResolvedPath "Utilities/SdkGen/EdFi.SdkGen.Console/csharp/EdFi.OdsApi.Sdk.sln")
+            $script:result += Invoke-Task "Restore-TestSdk-Packages" { Invoke-Restore-ApiSdk-Packages $sdkSolutionFile }
+            $script:result += Invoke-Task "Invoke-RebuildSolution" { Invoke-RebuildSolution $buildConfiguration "minimal" $sdkSolutionFile }
+
+            $nuspecFile = (Get-RepositoryResolvedPath "Utilities/SdkGen/EdFi.SdkGen.Console/EdFi.OdsApi.Sdk.nuspec")
+            $suffix = Get-ValueOrDefault $teamCityParameters['odsapi.package.suffix'] ".Suite3"
+            Write-Host "Updating package id and title to EdFi$suffix.OdsApi.TestSdk"
+            (Get-Content -path $nuspecFile -Raw) | ForEach-Object {
+                $_.replace("<id>EdFi.OdsApi.Sdk</id>","<id>EdFi$suffix.OdsApi.TestSdk</id>").replace("<title>EdFi.OdsApi.Sdk</title>","<title>EdFi$suffix.OdsApi.TestSdk</title>")
+            } | Set-Content -Path $nuspecFile
+            (Get-Content -path $nuspecFile -Raw) | ForEach-Object {
```

Comment:
This small piece code is repeated as above 3 lines

Reply:
It is repeated twice to cover both instances of how the nuspec file contents could need to be changed. The two paths covered are:
1. If we are only generating the test package, then the nuspec file would not have been modified yet and the id and title inside the nuspec will still be what is checked into source control, which is EdFi.OdsApi.Sdk. So we replace that with EdFi.Suite3.OdsApi.TestSdk
2. If we generate both api sdk package and test sdk package, then the nuspec file would have been modified for the api sdk package to have an id and title of EdFi.Suite3.OdsApi.Sdk, so we need to rename that to EdFi.Suite3.OdsApi.TestSdk

but looking at the code now. I think the two passes could be combined to something like this, where it does the 4 replaces together:
```
            (Get-Content -path $nuspecFile -Raw) | ForEach-Object {
                $_.replace("<id>EdFi.OdsApi.Sdk</id>","<id>EdFi$suffix.OdsApi.TestSdk</id>").replace("<title>EdFi.OdsApi.Sdk</title>","<title>EdFi$suffix.OdsApi.TestSdk</title>").replace("<id>EdFi$suffix.OdsApi.Sdk</id>","<id>EdFi$suffix.OdsApi.TestSdk</id>").replace("<title>EdFi$suffix.OdsApi.Sdk</title>","<title>EdFi$suffix.OdsApi.TestSdk</title>")
            } | Set-Content -Path $nuspecFile
```

Reply:
I have pushed up a change to do the four replace calls in one line with one loop now.